src/internal/packager2/layout/import.go

```diff
@@ -147,12 +147,23 @@ func resolveImports(ctx context.Context, pkg v1alpha1.ZarfPackage, packagePath,
 	}
 
 	pkg.Components = components
-	pkg.Variables = slices.CompactFunc(variables, func(l, r v1alpha1.InteractiveVariable) bool {
-		return l.Name == r.Name
-	})
-	pkg.Constants = slices.CompactFunc(constants, func(l, r v1alpha1.Constant) bool {
-		return l.Name == r.Name
-	})
+	// Keep only first occurrence of each variable
+	for _, v := range variables {
+		if slices.IndexFunc(pkg.Variables, func(existing v1alpha1.InteractiveVariable) bool {
```

Comment:
Agree with your assessment - the variables and constants are not used at scale to be a determinant consideration. 

For fun - at fist glance I want to reach for a `map[string]bool` here for uniqueness which I believe would O(n)?

Reply:
Yeah I originally was going to do that, but since maps are unordered, I didn't want the zarf.yaml to not be fully reproducible

Reply:
The map would only be for uniqueness checks though yes? Storage of the variable would still be appended to the `pkg.variables` I believe?

Reply:
You are correct, I just updated it to use maps instead